ecosystem/typescript/sdk_v2/src/api/general.ts

```diff
@@ -0,0 +1,114 @@
+import { getBlockByHeight, getBlockByVersion, getLedgerInfo, getTableItem, view } from "../internal/general";
+import { Block, LedgerInfo, LedgerVersion, MoveValue, TableItemRequest, ViewRequest } from "../types";
+import { AptosConfig } from "./aptos_config";
+
+/**
+ * A class to query all `General` Aptos related queries
+ */
+export class General {
+  readonly config: AptosConfig;
+
+  constructor(config: AptosConfig) {
+    this.config = config;
+  }
+
+  /**
+   * Queries for the Aptos ledger info
+   *
+   * @returns Aptos Ledger Info
+   *
+   * @example An example of the returned data
+   * ```
+   * {
+   * "chain_id": 4,
+   * "epoch": "8",
+   * "ledger_version": "714",
+   * "oldest_ledger_version": "0",
+   * "ledger_timestamp": "1694695496521775",
+   * "node_role": "validator",
+   * "oldest_block_height": "0",
+   * "block_height": "359",
+   * "git_hash": "c82193f36f4e185fed9f68c4ad21f6c6dd390c6e"
+   * }
+   * ```
+   */
+  async getLedgerInfo(): Promise<LedgerInfo> {
+    const data = await getLedgerInfo({ aptosConfig: this.config });
+    return data;
+  }
+
+  /**
+   * Queries for the chain id
+   *
+   * @returns The chain id
+   */
+  async getChainId(): Promise<number> {
+    const result = await this.getLedgerInfo();
+    return result.chain_id;
+  }
+
+  /**
+   * Queries for block by transaction version
+   *
+   * @param version Ledger version to lookup block information for
+   * @param options.withTransactions If set to true, include all transactions in the block
+   *
+   * @returns Block
+   */
+  async getBlockByVersion(args: { blockVersion: number; options?: { withTransactions?: boolean } }): Promise<Block> {
+    const block = await getBlockByVersion({ aptosConfig: this.config, ...args });
+    return block;
+  }
+
+  /**
+   * Get block by block height
+   *
```

Comment:
We will need to do a docs pass later for consistency etc.

e.g. The first one says Queries, the second says get.